src/Source/DoctrineODMQuerySourceIterator.php

```diff
@@ -139,20 +144,52 @@ public function getDateTimeFormat()
     }
 
     /**
-     * @param $value
+     * @return array
+     */
+    public function getDateFormats()
+    {
+        return $this->dateFormats;
+    }
+
+    /**
+     * @param array $dateFormats
+     */
+    public function setDateFormats(array $dateFormats = array())
+    {
+        $this->dateFormats = $dateFormats;
+    }
+
+    /**
+     * @param mixed        $target
+     * @param PropertyPath $propertyPath
      *
-     * @return null|string
+     * @return mixed
      */
-    protected function getValue($value)
+    protected function getValue($target, PropertyPath $propertyPath)
     {
+        $value = $this->propertyAccessor->getValue($target, $propertyPath);
         if (is_array($value) || $value instanceof \Traversable) {
             $value = null;
         } elseif ($value instanceof \DateTime) {
-            $value = $value->format($this->dateTimeFormat);
+            $value = $this->getFormattedDateForField($value, $propertyPath);
         } elseif (is_object($value)) {
             $value = (string) $value;
         }
```

Comment:
This if… elseif… elseif… chain is an OCP violation. There should be a new interface called SomethingFormatter, with one implementation per type and a default implementation.
Could look like

```
interface SomethingFormatterInterface
{
    public function supports($data);
    public function format($value, $propertyPath);
}
```

That would remove a lot of code duplication. If you do this, then I think your idea can also be implemented afterwards.